src/modules/cmdpal/Microsoft.CmdPal.UI/ShellPage.xaml.cs

```diff
@@ -95,4 +103,19 @@ public void Receive(PerformCommandMessage message)
             // TODO logging
         }
     }
+
+    public void Receive(ShowDetailsMessage message)
+    {
+        DetailsContent.ViewModel.Details = message.Details;
```

Comment:
This would be the last thing to sort out, but the Details control, can also just listen to this message and set it there directly in its own code.